api/src/controllers/emailNotificationController.ts

```diff
@@ -45,20 +47,19 @@ export const sendMentionEmailNotification = async (
 ) => {
   let senderName: string | undefined = ANONYMOUS_USER;
   if (!anonymous) {
-    const sender = await UserRepository.findUserById(senderId);
-    senderName = sender?.name;
+    const sender: User = await UserRepository.findUserById(senderId);
+    senderName = sender.name;
   }
 
-  const targetUsers = targetUserIds.map(async (mentionedUserId) => {
-    const user = await UserRepository.findUserById(mentionedUserId);
-    return user ? user.email : null;
+  const targetUsers: Promise<string>[] = targetUserIds.map(async (mentionedUserId) => {
+    const user: User = await UserRepository.findUserById(mentionedUserId);
+    return user.email;
   });
 
-  const targetUserEmails = (await Promise.all(targetUsers)).filter((email) => email !== null) as string[];
-
-  const project = await ProjectRepository.findProjectById(projectId);
-  const projectName = project?.name;
+  const targetUserEmails: string[] = (await Promise.all(targetUsers)).filter((email) => email !== null) as string[];
```

Comment:
I don't think the `email !== null` makes sense anymore. Because the `targetUsers` should never have any null values. Resolving the promises should be enough. also let's rename `targetUsers` to `targetUserEmailPromises`. `targetUsers` is misleading as those aren't users instead string promises. 
I would probably do sth like 

```
const targetUsersEmailPromises: Promise<string>[] = targetUserIds.map(async (targetUserId) => {
     const user: User = await UserRepository.findUserById(taregtUserId);
     return user.email;
   });
const targetUserEmails: string[] = await Promise.all(targetUsersEmailPromises)
```

Reply:
Right, will fix

Reply:
On second look, users can have null values if they did not complete registration.

Reply:
right good catch. we should force the users for email registration but nothing to do in this PR. Can we probably still rename the vars? Also I think we can get rid of the `user` var we can directly do `await UserRepository.findUserById(taregtUserId).email`

Reply:
Will need a bracket around the await like (await UserRepository.findUserById(taregtUserId)).email. I don't like it, if you want it, I can make the change.

Reply:
I think the change would it cleaner as we don't need the extra var and return statement. We can achieve the same thing in one line.